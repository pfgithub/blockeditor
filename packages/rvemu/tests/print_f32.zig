const std = @import("std");
const lib = @import("lib/start.zig");

// TODO: force debug build
// in releasesafe it prints 25

pub const panic = lib.panic;
export fn _start() noreturn {
    var buf: [30]u8 = undefined;
    var val: f32 = 25.0;
    _ = &val;
    var slice = std.fmt.bufPrint(&buf, "Print 25.0 ? {d}", .{val}) catch "[ERR]";
    _ = lib.syscall2(lib.Sys.print, @intFromPtr(slice.ptr), slice.len);

    var val2: f32 = 10.0;
    val = 1.0;
    _ = &val2;
    slice = std.fmt.bufPrint(&buf, "1.0 < 10.0 ? {}", .{val < val2}) catch "[ERR]";
    _ = lib.syscall2(lib.Sys.print, @intFromPtr(slice.ptr), slice.len);

    _ = lib.syscall1(lib.Sys.exit, 0);
    unreachable;
}
