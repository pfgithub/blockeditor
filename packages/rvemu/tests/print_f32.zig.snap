[LOG] Print 25.0 ? 25
[LOG] 1.0 < 10.0 ? true
Exited with code 0
